src/ATen/function_wrapper.py

```diff
@@ -365,45 +418,35 @@ def process_option(option, output_options):
                                                         else 'size' if broadcast_dims else 'outplace')
             option['broadcast_modified_actuals'] = ['b_' + y if 'b_' + y in option['broadcast_returns'] else y
                                                     for y in option['actuals']]
-
-            env = nested_dict(option, top_env)
-            top_env['type_method_declarations'].append(
-                TYPE_METHOD_DECLARATION_BROADCAST.substitute(env))
             top_env['type_method_definitions'].append(
                 TYPE_METHOD_DEFINITION_BROADCAST.substitute(env))
 
+        method_of = ['Type']
         if is_method:
             top_env['tensor_method_declarations'].append(
                 TENSOR_METHOD_DECLARATION.substitute(env))
             top_env['tensor_method_definitions'].append(
                 TENSOR_METHOD_DEFINITION.substitute(env))
+            method_of.append('Tensor')
+
+        if is_namespace_function:
+            option['inferred_type'] = 'infer_type({})'.format(first_tensor)
+            top_env['function_declarations'].append(
+                FUNCTION_DECLARATION.substitute(env))
+            top_env['function_definitions'].append(
+                FUNCTION_DEFINITION.substitute(env))
+            method_of.append('namespace')
+
+        if option['canonical']:
```

Comment:
Is this only exporting Declaration.yaml entries for canonical functions? I need the non-canonical entries to be in Declarations.yaml as will for the aten caffe2 operator. Generally, Declarations.yaml should document what the full public API is.